packages/react/src/components/textarea/textarea.tsx

```diff
@@ -1,29 +1,42 @@
 import { c, classy, m, TextareaProps as BaseProps } from '@onfido/castor';
 import React from 'react';
+import { FieldLabelWrapper } from '../../internal';
 import { withRef } from '../../utils';
 
+const idPrefix = 'castor_textarea';
+let idCount = 0;
+
 export const Textarea = withRef(
   (
     {
+      id = `${idPrefix}_${idCount++}`,
       resize = 'vertical',
       rows = 3,
       invalid,
+      children,
       className,
       style,
       ...restProps
     }: TextareaProps,
     ref: TextareaProps['ref']
   ): JSX.Element => (
-    <textarea
-      {...restProps}
-      ref={ref}
-      rows={rows}
-      className={classy(c('textarea'), m({ invalid }), className)}
-      style={{ ...style, resize }}
-    />
+    <FieldLabelWrapper id={id}>
+      {{
+        children,
+        element: (
+          <textarea
+            {...restProps}
+            ref={ref}
+            id={id}
+            rows={rows}
+            className={classy(c('textarea'), m({ invalid }), className)}
+            style={{ ...style, resize }}
```

Comment:
Now this gets tricky again.
Should we apply `style` and `className` to the parent `<label>` like we do with Checkbox and Radio?
We should at the very least apply `data-` and `aria-` attributes, I suppose.

Reply:
Here comes the annoying part... `FieldLabelWrapper` either wraps with `FieldLabel` _or_ renders input/textarea component directly.

For such case these two (`className` and `style`) should go back to input/textarea component, but then `cloneElement` should be used to pass the props. Which I know you don't like.

Reply:
Argh. Or we have ternaries to apply them in either or.

Let's leave it like this for now, I suppose further control over `FieldLabel` is always possible with the alternate rendering option:
```ts
<FieldLabel data-test="foo">
  <Input data-test="bar" />
</FieldLabel>
```

Reply:
Checkbox/Radio had issue because of an indicator, and original `<input>` hidden. This should not be problematic, but can always revisit if we spot something.